src/java.desktop/macosx/classes/sun/lwawt/macosx/CAccessible.java

```diff
@@ -182,7 +182,8 @@ public void propertyChange(PropertyChangeEvent e) {
 
                     // Do send check box state changes to native side
                     if (thisRole == AccessibleRole.CHECK_BOX) {
-                        if (newValue != null && !newValue.equals(oldValue)) {
+                        if ((newValue != null && !newValue.equals(oldValue)) ||
+                                oldValue != null && !oldValue.equals(newValue)) {
```

Comment:
```suggestion
                        if (!Objects.equals(newValue, oldValue)) {
```
This handles all the cases, including `null` values.